### DISCUSSION TITLE
Auto size columns width to fit available space

### DISCUSSION THREAD

**Discussion by vitolpoc** [infinite-table/infinite-react#11] [Ideas]
Provide a way to dynamically and proportionally redistribute available width to the columns taking into account the column’s definition (min width, width, max width) and distribute the available space to the columns or display a horizontal scrollbar if not enough width, similar to ag-grid’s sizeColumnToFit()

**Comment by roblotter** [comment]
We have an issue on this #9  and we're actively working on it

**Reply by roblotter** [reply]
https://infinite-table.com/docs/latest/learn/columns/fixed-and-flexible-size should be what you've been waiting for. 

Also make sure you check out https://infinite-table.com/docs/latest/reference#viewportReservedWidth for adjusting the reserved space, in case you need it - for advanced use-cases

**Reply by vitolpoc** [reply]
Version: 0.0.10
It's a step but not answering the full need.

Indeed, the size of the columns are either fixed either flexible. What I expected is that we could compose the width with the flex properties.

For instance:

```
const columnSizing: InfiniteTablePropColumnSizing = {
  "group-by-country": { width: 100 },
  city: { flex: 1, width: 50 },
  firstName: { flex: 1, width: 100, minWidth: 100 },
  lastName: { flex: 1, width: 150 },
  salary: { flex: 1 }
};
```
then if the container is larger than the total width then the available space would be redistributed according to the width of the columns, `firstName` would be smaller than the `lastName` column.
With the current solution, if I'm not wrong, we can only have relative width between flex columns not with the fixed width columns.

I tried to quickly illustrate the expected behavior in this sandbox: https://codesandbox.io/s/infinite-table-autosize-width-flex-tjrr8?file=/src/App.tsx .
![image](https://user-images.githubusercontent.com/42170298/148579060-eaf2e1ef-e049-4431-bbbd-90c5dd5c2925.png)


Nevertheless, this is not a high priority 

Also, it would be great if, when we have groups and `group-by-<field>` is not defined in columnSizing but `<field>` is, then the group size could be set by default to the value given for the `<field>`. This will allow to have only one definition (no matter if it's a group or not) and would be more convenient when the configuration of the groups are dynamic (as a user input for instance).

**Reply by roblotter** [reply]
I get your point. For the flex, this would mean implementing `flex-basis`. Agree it could be useful for certain very advanced situations, and we want to address this in the future, but it's not a priority right now. I've added an issue, so we have it logged #32

Your second proposal is also a nice one, so added a new issue #33. Definitely a nice touch to have more useful defaults. 👍
